cli/src/commands/fix.rs

```diff
@@ -185,39 +191,47 @@ pub(crate) fn cmd_fix(
             }
         }
 
-        // Also fix any new paths that were changed in this commit.
+        // There are four behaviors for the combinations of specifying or omitting the
+        // --include-unchanged-files flag or file arguments. The flag determines whether
+        // we diff the revision against its parents or just fix the entire tree. In
+        // either case, the file arguments filter that set of files.
         let tree = commit.tree()?;
-        let parent_tree = commit.parent_tree(tx.repo())?;
-        // TODO: handle copy tracking
-        let mut diff_stream = parent_tree.diff_stream(&tree, &matcher);
-        async {
-            while let Some(TreeDiffEntry {
-                path: repo_path,
-                values,
-            }) = diff_stream.next().await
-            {
-                let (_before, after) = values?;
+        let tree_values: Vec<(RepoPathBuf, Merge<Option<TreeValue>>)> = async {
+            if args.include_unchanged_files {
+                MergedTree::resolved(Tree::empty(tx.repo().store().clone(), RepoPathBuf::root()))
+            } else {
+                commit.parent_tree(tx.repo())?
+            }
+            .diff_stream(&tree, &matcher)
+            .map(|entry| {
+                // TODO: handle copy tracking by fixing all destination files, and only the
+                // source files that were otherwise modified. This will also need to extend to
+                // changed line computation once we support that.
                 // Deleted files have no file content to fix, and they have no terms in `after`,
                 // so we don't add any tool inputs for them. Conflicted files produce one tool
                 // input for each side of the conflict.
-                for term in after.into_iter().flatten() {
-                    // We currently only support fixing the content of normal files, so we skip
-                    // directories and symlinks, and we ignore the executable bit.
-                    if let TreeValue::File { id, executable: _ } = term {
-                        // TODO: Skip the file if its content is larger than some configured size,
-                        // preferably without actually reading it yet.
-                        let tool_input = ToolInput {
-                            file_id: id.clone(),
-                            repo_path: repo_path.clone(),
-                        };
-                        unique_tool_inputs.insert(tool_input.clone());
-                        paths.insert(repo_path.clone());
-                    }
+                entry.values.map(|(_before, after)| (entry.path, after))
+            })
+            .try_collect()
+            .await
```

Comment:
Looks like this async block can be removed since you're just awaiting a single value and immediately blocking on it.